app/routes.js

```diff
@@ -33,24 +33,26 @@ function add_route(route, router) {
   router[route.method](route.pattern, route.handler);
 }
 
-
-exports.url_for = function (route_name, args = {}) {
-
-  let route = routes[route_name];
+exports.url_for = function (route_name, {params = {}, fragment} = {}) {
```

Comment:
On second thought, I think this change should be reverted. It makes `url_for` less easy to use and we only use fragments in 3 places. Instead use `${url_for('handler.name', { id: ... })}#${escape(fragment)}`.

Reply:
😱  - Actually makes sense, I'll have a look.

Reply:
FYI: I tried to move commits around and/or revert that commit but it's been a pain so it seems I'll go for the simple option of merging this and the other PR and "revert" this `url_for()` changes manually 😞

Reply:
(as discussed offline few minutes ago)